nvim-oxi/src/api/vimscript.rs

```diff
@@ -1 +1,51 @@
+use nvim_types::{Array, Error};
+use nvim_types::{Object, ObjectData, ObjectType, String as NvimString};
 
+use super::ffi::vimscript::*;
+use crate::lua::LUA_INTERNAL_CALL;
+use crate::object::FromObject;
+use crate::Result;
+
+pub fn exec(src: &str, output: bool) -> Result<String> {
+    let src = NvimString::from(src);
+    let mut err = Error::new();
+    let output = unsafe {
+        nvim_exec(LUA_INTERNAL_CALL, src.non_owning(), output.into(), &mut err)
+    };
+    err.into_err_or_else(move || output)
+        .and_then(|output| output.into_string().map_err(From::from))
+}
+
+pub fn command(command: &str) -> Result<()> {
+    let command = NvimString::from(command);
+    let mut err = Error::new();
+    unsafe { nvim_command(command.non_owning(), &mut err) };
+    err.into_err_or_else(|| ())
+}
+
+pub fn eval<Value>(expr: &str) -> Result<Value>
+where
+    Value: FromObject,
+{
+    let expr = NvimString::from(expr);
+    let mut err = Error::new();
+    let output = unsafe { nvim_eval(expr.non_owning(), &mut err) };
+    err.into_err_or_flatten(|| Value::from_obj(output))
+}
+
+// TODO: accept tuple for args
+pub fn call_function<Ret>(func: &str, args: Vec<Object>) -> Result<Ret>
```

Comment:
We can implement `ToObject` for tuples `(A, B, ...)` where `(A, B, ..)` are all `ToObject` by converting the tuple to an `Array`. You can look [here](https://github.com/noib3/nvim-oxi/blob/a391cf154c660b665a87f59ce4322878976ac8a6/nvim-oxi/src/lua/pushable.rs#L92-L128) for a quick macro to reduce verbosity.